spec/struct.dd

```diff
@@ -1373,6 +1373,237 @@ $(H3 $(LNAME2 implicit-copy-constructors, Implicit Copy Constructors))
 
     $(P If the generated copy constructor fails to type check, it will receive the `@disable` attribute.)
 
+$(H2 $(LEGACY_LNAME2 StructMoveConstructor, struct-move-constructor, Struct Move Constructors))
+
+    $(P Move constructors are very much like copy constructors. The difference is that copy
+    constructors make a copy of the original, while move constructors move the contents of
+    the original, and the lifetime of the original ends.)
+
+    $(NOTE Do not use postblits in the same struct with move constructors.)
+
+    $(COMMENT $(P A `struct` that defines a move constructor
+    is not $(RELATIVE_LINK2 POD, POD).))
+    $(P If a move constructor is declared, also declare a copy constructor.)
+
+    $(P A constructor declaration is a move constructor declaration if it meets
+    the following requirements:)
+
+    $(UL
+    $(LI It takes exactly one parameter without a
+    $(DDSUBLINK spec/function, function-default-args, default argument),
+    followed by any number of parameters with default arguments.)
+
+    $(LI Its first parameter is not a
+    $(DDSUBLINK spec/function, ref-params, `ref` parameter).)
+
+    $(LI The type of its first parameter is the same type as
+    $(DDSUBLINK spec/type, typeof, `typeof(this)`), optionally with one or more
+    $(DDLINK spec/const3, Type Qualifiers, type qualifiers) applied to it.)
+
+    $(LI It is not a
+    $(DDSUBLINK spec/template, template_ctors, template constructor declaration).)
+    )
+
+    ---
+    struct A
+    {
+        this(ref return scope A rhs) {}                    // copy constructor
+        this(return scope A rhs) {}                        // move constructor
+        this(return scope const A rhs, int b = 7) {}       // move constructor with default parameter
+    }
+    ---
+
+    $(P The move constructor is type checked as a normal constructor.)
+
+    $(P The move constructor's first parameter only accepts rvalues. An
+    lvalue can be coerced into being an rvalue using
+    $(DDSUBLINK spec/expression, RvalueExpression, `__rvalue(Expression)`).)
+
+    $(P If a move constructor is defined, implicit calls to it will be inserted
+    in the following situations:)
+
+    $(OL
+    $(LI When a variable is explicitly initialized:)
+
+    $(SPEC_RUNNABLE_EXAMPLE_RUN
```

Comment:
All the added runnable/compilable examples in this PR fail to compile on run.dlang.org due to the compiler version it has:

> Error: `struct A` may not define both a rvalue constructor and a copy constructor

Reply:
It was awaiting the compiler PR.